### PR TITLE
Add Qwen support

### LINKED ISSUE #840: Qwen support
It would be very helpful if lit-gpt provides support for the new models Qwen (https://huggingface.co/Qwen). They have strong performance compared to other open source models.

### PR DESCRIPTION
**Pull request by chenht2026**
It works. 
Qwen's tokenizer is based on tiktoken, I add the tokenizer(tokenization_qwen.py) from its huggingface repo without any revision. This make the code a little complicate, so maybe do not merge.

May be some one needs it. 

Closes #840

### CODE REVIEW

**Comment by lantiga** [comment]
Thank you @chenht2026, sorry for the wait. A few of us took a break :-)

File: lit_gpt/config.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ class Config:
     rotary_percentage: float = 0.25
     parallel_residual: bool = True
     bias: bool = True
+    # just for Qwen
+    is_Qwen: Optional[bool] = None
```
Comment (lantiga) [inline lit_gpt/config.py:31]: I think we should avoid this in favor of something that characterizes Qwen, like having bias only in `c_attn`.
For the time being we could rename this as `attn_bias`, and then in the future turn `bias` into a `Option[bool, List[str]]` if there's a need for it.

**Review by lantiga — COMMENTED** [review]
Thanks for the contributions! One comment on the config flag

File: lit_gpt/tokenization_qwen.py [hunk 2]
```diff
@@ -0,0 +1,276 @@
+# Copyright (c) Alibaba Cloud.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
```
[the remark this answers is not in the document]
Comment (chenht2026) [reply lit_gpt/tokenization_qwen.py:4]: I just copy it from their huggingface repo.   [tokenization_qwen.py](https://huggingface.co/Qwen/Qwen-7B/blob/main/tokenization_qwen.py)

File: tutorials/download_Qwen.md
[the remark this answers is not in the document]
Comment (chenht2026) [reply tutorials/download_Qwen.md:1]: Chinese and English. I'll add it.

**Comment by samuelazran** [comment]
> It works. Qwen's tokenizer is based on tiktoken, I add the tokenizer(tokenization_qwen.py) from its huggingface repo without any revision. This make the code a little complicate, so maybe do not merge.
> 
> May be some one needs it.
> 
> Closes #840

Would it work with Qwen 2 (Qwen/Qwen1.5-7B-Chat)? if not what needs to be added?
Qwen1.5 improve a lot on its predecessor in performance.
https://huggingface.co/Qwen/Qwen1.5-7B-Chat/tree/main

**Comment by rasbt** [comment]
Hey just pinging to see if you are still interested in pursuing this PR. Personally, I think it'd be awesome to support the Qwen models (1.5 and especially 2) in LitGPT. There have been some improvements in the tokenizer in LitGPT recently that could now make this more easily possible.

Btw if rebasing here based on the main branch (which changed a lot) is too messy, you could also just open a fresh PR.

**Comment by Andrei-Aksionov** [comment]
Closing as it was added in #1834

**Pull request closed without merging**